app/controllers/mixins/ems_common/angular.rb

```diff
@@ -80,14 +81,18 @@ def realtime_authentication_check(verify_ems = nil)
         verify_ems.authentication_check(params[:cred_type], :save => false, :database => params[:metrics_database_name])
       end
 
+      def authentication_check(ems_type)
+        if %w[ems_cloud ems_infra].include?(params[:controller])
+          ems_type.validate_credentials_task(get_task_args(ems_type), session[:userid], params[:zone])
+        else
+          realtime_authentication_check(ems_type.new)
+        end
+      end
+
       def create_ems_button_validate
         @in_a_form = true
         ems_type = model.model_from_emstype(params[:emstype])
-        result, details = if %w[ems_cloud ems_infra].include?(params[:controller])
-                            ems_type.validate_credentials_task(get_task_args(ems_type), session[:userid], params[:zone])
-                          else
-                            realtime_authentication_check(ems_type.new)
-                          end
+        result, details = authentication_check(ems_type)
```

Comment:
TODO if you don't modify the password there is no `params[:default_password]` so https://github.com/ManageIQ/manageiq-ui-classic/pull/6667/files#diff-002200c01a8a53ff2a4439390e6bc3c4R125 isn't working.